iothub/device/tests/ConnectionString/IotHubConnectionStringBuilderTests.cs

```diff
@@ -41,6 +43,32 @@ public void IotHubConnectionStringBuilder_ParamConnectionString_ParsesDeviceId()
             csBuilder.DeviceId.Should().Be(DeviceId);
         }
 
+        [TestMethod]
+        [DoNotParallelize]
+        public void IotHubConnectionStringBuilder_ParamConnectionString_ParsesCultureDeviceId()
+        {
+            const string deviceId = "IoTDeviceCheck";
+
+            CultureInfo savedCultureInfo = Thread.CurrentThread.CurrentCulture;
+            CultureInfo[] allCultures = CultureInfo.GetCultures(CultureTypes.AllCultures);
+
+            try
+            {
+                foreach (CultureInfo culture in allCultures)
+                {
+                    Console.WriteLine($"Testing culture {culture}");
+                    Thread.CurrentThread.CurrentCulture = culture;
+                    var connectionString = $"HostName={HostName};DeviceId={deviceId};SharedAccessKey={SharedAccessKey}";
+                    var csBuilder = IotHubConnectionStringBuilder.Create(connectionString);
+                    csBuilder.DeviceId.Should().Be(deviceId);
```

Comment:
If it fails will we know what culture caused it?

Reply:
Good question. Yes because of the Console.WriteLine above, but I can add a comment to the assert to make it easier to find.